spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -1353,7 +1353,7 @@ protected void invokeListener(Channel channel, Message message) throws Exception
 	 * @throws Exception if the shared connection still can't be established
 	 */
 	protected void handleStartupFailure(BackOffExecution backOffExecution) throws Exception {
-		long recoveryInterval = backOffExecution.nextBackOff();
+		long recoveryInterval = backOffExecution != null ? backOffExecution.nextBackOff() : DEFAULT_RECOVERY_INTERVAL;
```

Comment:
Regarding this one - `ListenFromAutoDeleteQueueTests.testAutoDeclareFalse()`:
```
BlockingQueueConsumer consumer = mock(BlockingQueueConsumer.class);
doThrow(RuntimeException.class).when(consumer).start();
when(listenerContainer.createBlockingQueueConsumer()).thenReturn(consumer);
```
But `SimpleMessageListenerContainer.createBlockingQueueConsumer()` has this piece of code:
```
consumer.setBackOffExecution(this.recoveryBackOff.start());
```

Shouldn't we just fix that test-case instead of the strange NPE check here?